app/js/event/event.controller.js

```diff
@@ -13,13 +13,70 @@
     vm.events = [];
     vm.selectedEvent = null;
     vm.latestLocation = { latitude: 61.498180, longitude: 23.762195 };
+    vm.dragStartListener = null;
 
+    vm.renderEventsOnMap = renderEventsOnMap;
+    vm.showEventDetails = showEventDetails;
     vm.selectEvent = selectEvent;
     vm.toggleEventList = toggleEventList;
     vm.openBottomSheet = openBottomSheet;
 
+    $scope.$on('mapReady', onMapReady);
+
+    function onMapReady(e, map) {
+      e.preventDefault();
+      if (e.stopPropagation) {
+        e.stopPropagation();
+      }
+
+      vm.dragStartListener = map.addListener('dragstart', function() {
+          $scope.$apply(function() {
+            vm.selectedEvent = null;
+          });
+      });
+      vm.map = map;
+    }
+
+    function renderEventsOnMap(events) {
+      for (var i = 0; i < events.length; ++i) {
+        var event = events[i];
+        var previousEvent = events[i-1] || event;
+        var eventLoc = vm.map.createLatLng(Number(event.latitude),
+                                           Number(event.longitude));
+        var marker = vm.map.createMarker({
+          position: eventLoc,
+          title: event.name,
+          icon: 'img/icons/marker.event.png'
+        });
+
+        marker.addListener('click', showEventDetails(event, eventLoc, marker));
+        vm.map.showMarker(marker);
+
+        if (event.distanceToUser < previousEvent.distanceToUser) {
+          var nearestEventLoc = eventLoc;
+        }
+      }
+      if (nearestEventLoc) {
+        vm.map.panTo(nearestEventLoc);
```

Comment:
This function is called every time there's an location update and it forces the map to center somewhere. That's gonna be a pain in the ass when you just try to browse stuff  when on a bus and GPS keeps sending location events every once in a while